llama/grammar_test.go

```diff
@@ -0,0 +1,80 @@
+package llama
+
+import (
+	"bufio"
+	"bytes"
+	"strings"
+	"testing"
+)
+
+// https://github.com/ollama/ollama/issues/7978
+const issue7978JSONSchema = `{
+  "type": "object",
+  "properties": {
+    "steps": {
+      "type": "array",
+      "items": {
+	"type": "object",
+	"properties": {
+	  "explanation": { "type": "string" },
+	  "output": { "type": "string" }
+	},
+	"required": ["explanation", "output"],
+	"additionalProperties": false
+      }
+    },
+    "final_answer": { "type": "string" }
+  },
+  "required": ["steps", "final_answer"],
+  "additionalProperties": false
+}`
+
+func TestIssue7978(t *testing.T) {
```

Comment:
```suggestion
func TestJSONSchemaOrdering(t *testing.T) {
```

Reply:
a) Thank you for using JSON, not Json
b) It's specifically about key ordering, not all things that can be order
c) This test is very specific to the issue the name refers to, so I'd like to keep it named as-is, later, when/if we have more generic ordering tests, we can call it `TestSchemaToJSONPreservedKeyOrder(...)` or something like that.